hugo/content/guides/keywords-as-identifiers/_index.md

```diff
@@ -0,0 +1,164 @@
+                type: SemanticTokenTypes.class
+            });
+        }
+    }
+}
+```
+
+For all persons (`isPerson(...)` in line 7), we explicitly specify the semantic type for the token of their `'name'` property.
+Here, we use `SemanticTokenTypes.class` as semantic type.
+For your case, select a predefined type which fits your domain best.
+Since the name is used as cross-reference by greetings, a similar check and assignment of a semantic token type needs to be done for the `person` property of `Greeting` as well.
+
+After creating the semantic token provider, you need to register the `HelloWorldSemanticTokenProvider` in `hello-world-module.ts` in the following way:
+
+```ts
+export const HelloWorldModule: Module<HelloWorldServices, PartialLangiumServices & HelloWorldAddedServices> = {
+    lsp: {
+        SemanticTokenProvider: (services) => new HelloWorldSemanticTokenProvider(services)
+    },
+    validation: {
+        HelloWorldValidator: () => new HelloWorldValidator()
+    }
+};
+```
+
+Now rebuild and restart your application and test the improvements of the second step:
+
+![screenshot with fixed semantic token](fixed-2-token.png)
+
+The `HelloWorldSemanticTokenProvider` works, and you might see a different highlighting XOR you might not see any difference, e.g. "Hello" is still blue here. This leads us to the third step.
+
+
+## Step 3: Ensure that your editor styles the chosen semantic token type
+
+The __third step__ is to ensure, that your editor supports the assigned semantic tokens:
+Depending on your editor and the currently selected color theme, the semantic token type selected in `HelloWorldSemanticTokenProvider` might not be supported or didn't got a different color in the color theme.
+The easiest way to detect such problems is to change the current color theme and to try some others.
+Note, that VS Code allows to switch off semantic highlighting for all themes with the setting `editor.semanticHighlighting.enabled`.
+
+After switching from "Dark (Visual Studio)" to "Dark Modern" in VS Code, the example looks as expected.
+You can switch the current color theme in VS Code with `cmd + K` `cmd + T` (or via the menu: Code -> Settings... -> Theme -> Color Theme).
+
+![screenshot with supporting color theme](fixed-3-style-2.png)
+
+Now "Hello" is highlighted in purple if used as keyword, it's written in green if it is used as value for the name of a person.
+Another solution is to select a different semantic type for your token in step two.
+
+While __step one__ is mandatory to enable keywords as values in general,
+__step two__ improves the user experience of your language.
+While step one and step two can be handled in the LSP server once for your language, __step three__ highly depends on your editor and its color themes (in the LSP clients), which makes step three quite complicated to handle.
+
+
+Now you have learned, how to technically enable keywords as regular values for properties.
```

Comment:
IMO this comma is also superfluous/wrong, as there's neither a noun nor a verb in the subsequent part of the sentence.

Reply:
After checking again, I would say that the comma is wrong, since the sentence starts with the main clause and not with the subordinate clause.

I removed the comma.